docs/NewTargetSpecificNode.md

```diff
@@ -0,0 +1,93 @@
+
+There are mainly two advantanges of operator fusion:
+
+- Good locality and less memory access
+- Reduce kernel launch time in GPU and specialized accelerators
+
+In TensorRT, the convolution, bias and ReLU layers of various sizes can be combined into a single kernel called CBR. A simple analogy is making three separate trips to the supermarket to buy three items versus buying all three in a single trip.
+
+
+#### Data Layout Transformation
+
+Actually, a Tensor is a view of a block of memory. Besides a pointer to the memory, we also have to get some other descriptions of this block of memory, such as shape, stride and layout.
+
+Different layout leads to different implementation of the operator kernel on certain hardware. For example, `NCHW` is a common layout of the input tensor of convolution operator. In Intel CPU, the best performance is achieved after transforming layout `NCHW` to layout `nChw8c`. After Tensor Core was introduced to Volta GPU, layout `NCHW` is not more efficient than layout `NHWC`.
+
+We should take both fast operator kernel implementation and extra potential layout transformation into consideration to get better performance.
+
+### Steps
+
+There are mainly three steps to add new target specific node in Glow:
+
+1. Add a backend-specific Node `FusedAB` to `XSpecificNodes.h`, and a corresponding backend-specific Instruction `FusedAB` to `XSpecificInstrs.h`. Note that the `FusedABInst` needs to be marked with `autoIRGen()` so that the node is automatically IRGen'd to the instruction, as we currently do not support backend-specific IRGen.
+
+2. Add logic to `XBackend::transformPreLowering()` or `XBackend::transformPostLowering()` (or both) depending on when you want to do the transformation. This logic would look for the pattern of nodes you want to fuse (`A` with a single use by `B`), and replaces all uses of the result of B with the new backend-specific `FusedABNode`.
+
+3. Have your backend `X` implement `FusedABInst`. For example, for the OpenCL backend, this would mean adding a case to enqueue a kernel for the `FusedABInst` to `OpenCLFunction::execute()`, and then adding the corresponding kernel in `kernels.cl`.
+
+
+### Examples
+
+#### Operator Fusion for ReLU in CPU
+
+ReLU is max between zero and the input value. Glow lowers ReLU Node to two basic low-level linear algebra operator nodes, Splat Node and Max Node.
+
+Please refer to the document in [Backend](https://github.com/pytorch/glow/blob/master/docs/Backends.md#backend-specific-nodes-and-instructions) part for source code details on adding a new target specific CPUMaxSplatNode on CPU. 
+
+#### Data Layout Transformation for Conv Operator in OpenCL
+
+The default layout of convolution operator in Glow is `NHWC`. Since OpenCL fast convolution kernel itself has some issue with group > 1, we should convert layout `NHWC` to `NCHW` if the group > 1. Please refer to the [TODO](https://github.com/pytorch/glow/blob/master/lib/Backends/OpenCL/Transforms.cpp#L39) note. 
```

Comment:
I don't think it's important to refer to the TODO. You can just say here that OpenCL Conv is faster when in `NCHW`, but Glow's default layout is `NHWC`. So when we see a `Convolution` that we can use the fast kernel with, we transpose the inputs/output and replace the `Convolution` with a backend-specific `OCLConvolution` that expects/uses `NCHW`. Could also mention that the transposes mostly can get optimized away thanks to the high-level graph optimizations.

Reply:
Done